packages/terra-date-picker/src/DateRange.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { PropTypes } from 'react';
+import DatePicker from '../lib/DatePicker';
+
+const propTypes = {
+  endDate: PropTypes.oneOfType([PropTypes.object]),
+  startDate: PropTypes.oneOfType([PropTypes.object]),
+  onChange: PropTypes.func,
+};
+
+class DateRange extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: props.startDate,
+      endDate: props.endDate,
+    };
+    this.handleChangeStart = this.handleChangeStart.bind(this);
+    this.handleChangeEnd = this.handleChangeEnd.bind(this);
+  }
+
+  handleChange({ startDate, endDate }) {
+    let startDateForRange = startDate || this.state.startDate;
+    let endDateForRange = endDate || this.state.endDate;
+
+    if (startDateForRange.isAfter(endDateForRange)) {
+      const tempDate = startDateForRange;
```

Comment:
If you feel like it, this can use destructuring assignment: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

```
let startDateForRange = "march 20th";
let endDateForRange = "march 5th";

//  print initial values
console.log([startDateForRange, endDateForRange])

// swap values
[startDateForRange, endDateForRange] = [endDateForRange, startDateForRange]

//  print swapped values
console.log([startDateForRange, endDateForRange])
```

Up to you if you want to use it.